assert/http_assertions.go

```diff
@@ -163,3 +164,50 @@ func HTTPBodyNotContains(t TestingT, handler http.HandlerFunc, method, url strin
 
 	return !contains
 }
+
+// HTTP asserts that a specfied handler returns set of expected values given by HttpOptions.
+//
+//	assert.HTTP(t, myHandler, "www.google.com", nil, WithCode(200), WithBody("I'm Feeling Lucky"), WithRequestHeader(http.Header{"a": []string{"b"}}, WithExpectedBody(bytes.NewBuffer("c"))))
+//
+// Returns whether the assertion was successful (true) or not (false).
+func HTTP(t TestingT, handler http.HandlerFunc, method, url string, values url.Values, options ...HttpOption) bool {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+
+	b := &builder{}
+	for _, option := range options {
+		err := option(b)
+		if err != nil {
+			Fail(t, fmt.Sprintf("Failed to build http options, got error: %s", err))
+		}
+	}
+
+	w := httptest.NewRecorder()
+	req, err := http.NewRequest(method, url, b.body)
+	if b.err == nil && err != nil {
+		Fail(t, fmt.Sprintf("Failed to build test request, got error: %s", err))
+	} else if b.err != nil {
+		return err == b.err
+	}
+
+	req.Header = b.requestHeader
+	req.URL.RawQuery = values.Encode()
+	handler(w, req)
+	if w.Code != b.code {
+		Fail(t, fmt.Sprintf("Expected HTTP success status code for %q but received %d", url+"?"+values.Encode(), w.Code))
+		return false
+	}
```

Comment:
We can reuse the existing assertions. It already has a lot of diff logic.

```suggestion
	Equal(t, b.code, w.Code)
```

```
=== RUN   TestHttp
    main_test.go:15: 
        	Error Trace:	/Users/armahishi/dev/personal/forks/testify/example/main_test.go:15
        	Error:      	Expected HTTP success status code for "/?" but received 200
        	Test:       	TestHttp
--- FAIL: TestHttp (0.00s)
FAIL
FAIL	github.com/stretchr/testify/example	0.161s
FAIL
```

VS

```
=== RUN   TestHttp
    main_test.go:15: 
        	Error Trace:	/Users/armahishi/dev/personal/forks/testify/example/main_test.go:15
        	Error:      	Expected HTTP success status code for "/?" but received 200
        	Test:       	TestHttp
--- FAIL: TestHttp (0.00s)
FAIL
FAIL	github.com/stretchr/testify/example	0.161s
FAIL
```

Reply:
I see your point, however I wanted to be concise with the code style in the module, and there is not usage of `Equal` even though it is possible. 
As I am more sceptical to use module itself I think it is correct to not use `Equal` function as if it will be changed in future, you have to rework whole `http_assertions.go` module as it is dependend on that.
It is probably bad assumption but I think we can have a discussion about this and possibly make another PR with appropriate refactor of `http_assertions.go` module. :sweat_smile: